core/src/main/java/org/apache/cxf/staxutils/transform/InTransformReader.java

```diff
@@ -262,12 +262,22 @@ private void handleDefaultMode(QName name, QName expected) {
 
     private void handleDeepDrop() throws XMLStreamException {
         final int depth = getDepth();
-        while (depth != getDepth() || super.next() != XMLStreamConstants.END_ELEMENT) {
-            // get to the matching end element event
+        int read = 0;
+        while (depth != getDepth() || hasNext()) {
```

Comment:
think you can drop `depth != depth()` condition along side depth declaration.

Reply:
👍 thank you